elide-graphql/src/main/java/com/yahoo/elide/graphql/GraphQLEntityProjectionMaker.java

```diff
@@ -0,0 +1,410 @@
+package com.yahoo.elide.graphql;
+
+import com.yahoo.elide.ElideSettings;
+import com.yahoo.elide.core.EntityDictionary;
+import com.yahoo.elide.core.RelationshipType;
+import com.yahoo.elide.core.exceptions.InvalidEntityBodyException;
+import com.yahoo.elide.core.pagination.Pagination;
+import com.yahoo.elide.core.sort.Sorting;
+import com.yahoo.elide.request.Attribute;
+import com.yahoo.elide.request.EntityProjection;
+
+import graphql.language.Argument;
+import graphql.language.Document;
+import graphql.language.Field;
+import graphql.language.FragmentDefinition;
+import graphql.language.FragmentSpread;
+import graphql.language.OperationDefinition;
+import graphql.language.Selection;
+import graphql.language.SelectionSet;
+import graphql.language.Value;
+import graphql.parser.Parser;
+import lombok.extern.slf4j.Slf4j;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * This class converts a GraphQL query string into an Elide {@link EntityProjection} using {@link #make(String)}.
+ */
+@Slf4j
+public class GraphQLEntityProjectionMaker {
+    private final EntityDictionary entityDictionary;
+    private final ElideSettings elideSettings;
+
+    private final Map<String, List<EntityProjection>> fragmentToProjections = new HashMap<>();
```

Comment:
I have an idea.  What if you first translate the query to remove all fragments?  Then you don't need to keep all this state around when building the projections.  

It would also break up the code into smaller bits (that are mostly unrelated).